config/Project-Debug.xcconfig

```diff
@@ -0,0 +1,84 @@
+// Bundle Resources or Copy Files build phase, should be stripped of debugging symbols.
+// It does not cause the linked product of a target to be stripped—use
+// `STRIP_INSTALLED_PRODUCT` for that.
+COPY_PHASE_STRIP = NO
+
+// Enable Testability
+//
+// When this setting is activated, the product will be built with options appropriate for
+// running automated tests, such as making private interfaces accessible to the tests.
+// This may result in tests running slower than they would without testability enabled.
+ENABLE_TESTABILITY = YES
+
+// Generate Position-Dependent Code
+//
+// Faster function calls for applications. Not appropriate for shared libraries, which
+// need to be position-independent.
+GCC_DYNAMIC_NO_PIC = NO
+
+// Optimization Level
+//
+// Specifies the degree to which the generated code is optimized for speed and binary
+// size.
+//
+// * None: Do not optimize. [-O0]
+// With this setting, the compiler's goal is to reduce the cost of compilation and to
+// make debugging produce the expected results. Statements are independent—if you stop
+// the program with a breakpoint between statements, you can then assign a new value to
+// any variable or change the program counter to any other statement in the function and
+// get exactly the results you would expect from the source code.
+// * Fast: Optimizing compilation takes somewhat more time, and a lot more memory for a
+// large function. [-O1]
+// With this setting, the compiler tries to reduce code size and execution time,
+// without performing any optimizations that take a great deal of compilation time. In
+// Apple's compiler, strict aliasing, block reordering, and inter-block scheduling are
+// disabled by default when optimizing.
+// * Faster: The compiler performs nearly all supported optimizations that do not
+// involve a space-speed tradeoff. [-O2]
+// With this setting, the compiler does not perform loop unrolling or function
+// inlining, or register renaming. As compared to the `Fast` setting, this setting
+// increases both compilation time and the performance of the generated code.
+// * Fastest: Turns on all optimizations specified by the `Faster` setting and also
+// turns on function inlining and register renaming options. This setting may result in a
+// larger binary. [-O3]
+// * Fastest, Smallest: Optimize for size. This setting enables all `Faster`
+// optimizations that do not typically increase code size. It also performs further
+// optimizations designed to reduce code size. [-Os]
+// * Fastest, Aggressive Optimizations: This setting enables `Fastest` but also enables
+// aggressive optimizations that may break strict standards compliance but should work
+// well on well-behaved code. [-Ofast]
+// * Smallest, Aggressive Size Optimizations: This setting enables additional size
+// savings by isolating repetitive code patterns into a compiler generated function.
+// [-Oz]
+GCC_OPTIMIZATION_LEVEL = 0
```

Comment:
This is what I mean regarding the comments being useful but also noisy 😳 

This information is available [online](https://developer.apple.com/library/archive/documentation/DeveloperTools/Reference/XcodeBuildSettingRef/1-Build_Setting_Reference/build_setting_ref.html#//apple_ref/doc/uid/TP40003931-CH3-SW90) and in the Xcode Help:

<img width="437" alt="Screen Shot 2020-11-06 at 1 59 01 pm" src="https://user-images.githubusercontent.com/1218433/98321061-44d50280-2038-11eb-856e-28bb535f23ce.png">

Not having the comment would be a bit of extra friction to get to the meaning of a setting, but if it makes the file easier to digest and work with, it might be worth it.

Reply:
Yeah I agree that comments _that_ long will likely be too noisy for little benefits. It would be nice if it was only including the first line instead of the whole shebang.

But at that point I think having such lengthy comments is not worth it imho: we could always refer to either the official –even if not fully complete– Apple documentation, or to de-facto references like https://pewpewthespells.com/blog/buildsettings.html (†).  
I think it's better to use comments for manually documenting why we introduced some esoteric build setting for non-obvious specific cases, rather than using them to show the raw definition of the setting which can always be found somewhere else anyway.

_(†) Sam also has a couple of other articles [in her blog](https://pewpewthespells.com/blog/), about the Xcode build system, and how to use Build Settings efficiently, etc, that could be worth referencing btw._